src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Pass.cpp

```diff
@@ -14,11 +14,25 @@ namespace trajectory
 namespace orbit
 {
 
-Pass::Pass(const Pass::Type& aType, const Integer& aRevolutionNumber, const Interval& anInterval)
+Pass::Pass(
+    const Pass::Type& aType,
+    const Integer& aRevolutionNumber,
+    const Interval& anInterval,
+    const Instant& anInstantAtDescendingNode,
+    const Instant& anInstantAtNorthPoint,
+    const Instant& anInstantAtSouthPoint
+)
     : type_(aType),
       revolutionNumber_(aRevolutionNumber),
-      interval_(anInterval)
+      interval_(anInterval),
+      instantAtDescendingNode_(anInstantAtDescendingNode),
+      instantAtNorthPoint_(anInstantAtNorthPoint),
+      instantAtSouthPoint_(anInstantAtSouthPoint)
 {
+    if ((aType == Pass::Type::Complete) && (!anInstantAtNorthPoint.isDefined() || !anInstantAtSouthPoint.isDefined()))
```

Comment:
Should we check that `anInterval` and `anInstantAtDescendingNode` are defined too?